Simplenote/src/main/java/com/automattic/simplenote/viewmodels/RequestMagicLinkViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package com.automattic.simplenote.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.automattic.simplenote.di.IO_THREAD
+import com.automattic.simplenote.networking.SimpleHttp
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Named
+
+@HiltViewModel
+class RequestMagicLinkViewModel @Inject constructor(
+    private val simpleHttp: SimpleHttp,
+    @Named(IO_THREAD) private val ioDispatcher: CoroutineDispatcher,
+) : ViewModel() {
+    private val _magicLinkRequestUiState = MutableLiveData<MagicLinkRequestUiState>()
+    val magicLinkRequestUiState: LiveData<MagicLinkRequestUiState> get() = _magicLinkRequestUiState
+
+    fun requestLogin(username: String) = viewModelScope.launch(ioDispatcher) {
+        _magicLinkRequestUiState.postValue(MagicLinkRequestUiState.Loading(message = "Requesting Magic Link"))
```

Comment:
Can you make this localizable, and maybe say `Requesting log in link?` instead?

Reply:
Yes! I guess I will do the same for the others. There are a few of those 😅 . Thanks for pointing it out.

Reply:
Done